atomic_reactor/util.py

```diff
@@ -1128,12 +1128,29 @@ def get_primary_images(workflow):
 
 
 def split_module_spec(module):
-    m = re.match(r'^(.*)-([^-]+)-(\d{14})$', module)
+    # Current module naming guidelines are at:
+    # https://docs.pagure.org/modularity/development/building-modules/naming-policy.html
+    # We simplify the possible NAME:STREAM:CONTEXT:ARCH/PROFILE and only care about
+    # NAME:STREAM or NAME:STREAM:VERSION. PROFILE is not needed, and ARCH is determined by
+    # the architecture. CONTEXT may become important in the future, but we ignore it
+    # for now.
+    #
+    # Previously the separator was '-' instead of ':', which required hardcoding the
+    # format of VERSION to distinguish between HYPHENATED-NAME-STREAM and NAME-STREAM-VERSION.
+    # We support the old format for compatibility.
+    #
+    m = re.match(r'^([^:]+):([^:]+):([^:]+)$', module)
     if not m:
-        m = re.match(r'^(.*)-([^-]+)$', module)
-        if not m:
-            raise RuntimeError(
-                'Module specification should be NAME-STREAM or NAME-STREAM-VERSION')
+        m = re.match(r'^([^:]+):([^:]+)$', module)
+    if not m:
+        m = re.match(r'^(.+)-([^-]+)-(\d{14})$', module)
+    if not m:
+        m = re.match(r'^(.+)-([^-]+)$', module)
+    if not m:
+        raise RuntimeError(
+            'Module specification should be NAME:STREAM or NAME:STREAM:VERSION. ' +
+            '(NAME-STREAM and NAME-STREAM-VERSION supported for compatibility.)'
+        )
```

Comment:
This could be rewritten with a list of patterns and a for/break/else iterator. I don't think it would be much shorter, but it would be easier to extend in the future. 

The repeated if not m statements aren't wrong, but something about them bothers me.